core/lib/config/src/configs/genesis.rs

```diff
@@ -1,15 +1,135 @@
+            evm_emulator_hash: config.evm_emulator_hash,
+            l1_chain_id: config.l1_chain_id,
+            l2_chain_id: config.l2_chain_id,
+            fee_account: config.fee_account,
+            l1_batch_commit_data_generator_mode: config.l1_batch_commit_data_generator_mode,
+            custom_genesis_state_path: config.custom_genesis_state_path,
+            prover: PersistedGenesisProverConfig {
+                dummy_verifier: config.dummy_verifier,
+                snark_wrapper_vk_hash: config.snark_wrapper_vk_hash,
+                fflonk_snark_wrapper_vk_hash: config.fflonk_snark_wrapper_vk_hash,
+            },
+        })
+    }
+}
+
+impl TryFrom<PersistedGenesisConfig> for GenesisConfig {
+    type Error = ErrorWithOrigin;
+
+    fn try_from(config: PersistedGenesisConfig) -> Result<Self, Self::Error> {
+        Ok(Self {
+            protocol_version: Some(match config.genesis_protocol_semantic_version {
+                Some(ver) => ver,
+                None => {
+                    let minor = config.genesis_protocol_version.ok_or_else(|| {
+                        DeError::custom("Either genesis_protocol_version or genesis_protocol_semantic_version should be presented")
+                    })?;
+                    let minor: ProtocolVersionId = minor.try_into().map_err(|_| {
+                        DeError::invalid_value(
+                            Unexpected::Unsigned(minor.into()),
+                            &"protocol version ID",
+                        )
+                    })?;
+                    ProtocolSemanticVersion::new(minor, 0.into())
+                }
+            }),
+            genesis_root_hash: Some(config.genesis_root),
+            rollup_last_leaf_index: Some(config.genesis_rollup_leaf_index),
+            genesis_commitment: Some(config.genesis_batch_commitment),
+            bootloader_hash: Some(config.bootloader_hash),
+            default_aa_hash: Some(config.default_aa_hash),
+            evm_emulator_hash: config.evm_emulator_hash,
+            l1_chain_id: config.l1_chain_id,
+            l2_chain_id: config.l2_chain_id,
+            snark_wrapper_vk_hash: config.prover.snark_wrapper_vk_hash,
+            fee_account: config.fee_account,
+            dummy_verifier: config.prover.dummy_verifier,
+            l1_batch_commit_data_generator_mode: config.l1_batch_commit_data_generator_mode,
+            fflonk_snark_wrapper_vk_hash: config.prover.fflonk_snark_wrapper_vk_hash,
+            custom_genesis_state_path: config.custom_genesis_state_path,
+        })
+    }
+}
+
 /// This config represents the genesis state of the chain.
 /// Each chain has this config immutable and we update it only during the protocol upgrade
 #[derive(Debug, Clone, PartialEq)]
 pub struct GenesisConfig {
-    // TODO make fields non optional, once we fully moved to file based configs.
-    // Now for backward compatibility we keep it optional
+    // TODO make fields non optional?
```

Comment:
Actually, we moved to file based configs everywhere and we could remove this optional

Reply:
Would prefer to do this in a follow-up PR if possible.